pootle/core/checks/checker.py

```diff
@@ -0,0 +1,305 @@
+                # keep false-positive checks if check is active
+                unmute = (
+                    self.original_checks[name]['false_positive']
+                    and not self.keep_false_positives)
+                if unmute:
+                    self.unmute_list.append(name)
+                # if the check is valid remove from the list and continue
+                del self.original_checks[name]
+                continue
+
+            # the check didnt exist previously - so create it
+            self.check_qs.create(
+                unit_id=self.unit.id,
+                name=name,
+                message=self.check_failures[name]['message'],
+                category=self.check_failures[name]['category'])
+            updated = True
+
+        return updated
+
+
+class QualityCheckUpdater(object):
+
+    def __init__(self, check_names=None, translation_project=None,
+                 keep_false_positives=True):
+        """Refreshes QualityChecks for Units
+
+        :param check_names: limit checks to given list of quality check names.
+        :param translation_project: an instance of `TranslationProject` to
+            restrict the update to.
+        :param keep_false_positives: when set to `False`, it will activate
+            (unmute) any existing false positive checks.
+        """
+
+        self.check_names = check_names
+        self.translation_project = translation_project
+        self.keep_false_positives = keep_false_positives
+        self.stores = set()
+
+    @cached_property
+    def checks(self):
+        """Existing checks in the database for all units
+        """
+        checks = self.check_qs
+        check_keys = (
+            'id', 'name', 'unit_id',
+            'category', 'false_positive')
+
+        if self.check_names is not None:
+            checks = checks.filter(name__in=self.check_names)
+
+        all_units_checks = {}
+        for check in checks.values(*check_keys):
+            all_units_checks.setdefault(
+                check['unit_id'], {})[check['name']] = check
+        return all_units_checks
+
+    @cached_property
+    def check_qs(self):
+        """QualityCheck queryset for all units, restricted to TP if set
+        """
+        check_qs = QualityCheck.objects.all()
+
+        if self.translation_project is not None:
+            tp_pk = self.translation_project.pk
+            check_qs = check_qs.filter(
+                unit__store__translation_project__pk=tp_pk)
+        return check_qs
+
+    @cached_property
+    def units(self):
+        """Result set of Units, restricted to TP if set
+        """
+        units = Unit.simple_objects.all()
+        if self.translation_project is not None:
+            units = units.filter(
+                store__translation_project=self.translation_project)
+        return units
+
+    def clear_checks(self):
+        QualityCheck.delete_unknown_checks()
+
+    @lru_cache(maxsize=None)
+    def get_checker(self, tp):
```

Comment:
Also this method looks like it belongs to `pootle_translationproject` app rather than here.